generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -744,14 +754,17 @@ private RecordTemplateSpec generateRecord(RecordDataSchema schema)
     final List<NamedDataSchema> includes = schema.getInclude();
     for (NamedDataSchema includedSchema : includes)
     {
-      processSchema(includedSchema, null, null);
+      processSchema(includedSchema, null, null, skipDeprecatedField);
     }
 
     final Map<CustomInfoSpec, Object> customInfoMap = new IdentityHashMap<>(schema.getFields().size() * 2);
 
     for (RecordDataSchema.Field field : schema.getFields())
     {
-      final ClassTemplateSpec fieldClass = processSchema(field.getType(), recordClass, field.getName());
+      if (skipDeprecatedField && isDeprecated(field)) {
```

Comment:
Leave a comment here explaining that this affects generation recursively (i.e. it skips generating this field, its type, and types referenced within this type). It's not totally clear just looking at it.

Reply:
updated.